packages/pg-v5/commands/pull.js

```diff
@@ -125,19 +123,58 @@ const maybeTunnel = function * (herokuDb) {
   return herokuDb
 }
 
+function spawnPipe (pgDump, pgRestore) {
+  return new Promise((resolve, reject) => {
+    pgDump.stdout.pipe(pgRestore.stdin)
+    pgDump.on('close', (code) => {
+      if (code !== 0) {
+        reject(new Error(`pg_dump errored with ${code}`))
+      }
+      pgRestore.stdin.end()
+    })
+
+    pgRestore.on('close', (code) => {
+      if (code === 0) {
+        resolve(undefined)
+      } else {
+        reject(new Error(`pg_restore errored with ${code}`))
+      }
+    })
+  })
+}
+
 const run = co.wrap(function * (sourceIn, targetIn, exclusions) {
   yield prepare(targetIn)
 
   const source = yield maybeTunnel(sourceIn)
   const target = yield maybeTunnel(targetIn)
-
   const exclude = exclusions.map(function (e) { return '--exclude-table-data=' + e }).join(' ')
 
-  let password = p => p ? ` PGPASSWORD="${p}"` : ''
-  let dump = `env${password(source.password)} PGSSLMODE=prefer pg_dump --verbose -F c -Z 0 ${exclude} ${connstring(source, true)}`
-  let restore = `env${password(target.password)} pg_restore --verbose --no-acl --no-owner ${connstring(target)}`
+  let dumpFlags = ['--verbose', '-F', 'c', '-Z', '0', ...connArgs(source, true)]
+  if (exclude !== '') dumpFlags.push(exclude)
+
+  const dumpOptions = {
+    env: {
+      PGSSLMODE: 'prefer'
+    },
+    stdio: ['pipe', 'pipe', 2],
+    encoding: 'utf8',
+    shell: true
```

Comment:
would you mind testing if it works without this?

Reply:
@jdxcode Yes, sorry about that - you had asked earlier about this.  

I did try it out without `shell: true` and it fails with:
```
Error: spawn pg_dump ENOENT
    at Process.ChildProcess._handle.onexit (internal/child_process.js:231:19)
    at onErrorNT (internal/child_process.js:406:16)
    at process._tickCallback (internal/process/next_tick.js:63:19)
pg_restore: [custom archiver] could not read from input file: end of file
```

I can look into this some more.

Reply:
no need to investigate more, if we need it that's probably fine. My understanding is that `shell: true` does a lot of extra stuff behind the scenes that I'd rather not do, but if it only works with it enabled that's fine too.